### DISCUSSION TITLE
How to get bidirectional link between two documents from different repos.

### DISCUSSION THREAD

**Discussion by vinodreddy-g** [useblocks/sphinx-needs#1166] [Q&A]
I have recently started using sphinx-needs in our project. 

I build a requirements document in repo1 and generate the needs.json . Then use this needs.json in another repo with needs_external_needs and link it to derived requirements.

I would like to get bidirectional tracing between the requirements .
In the derived requirements i get the link since i use the needs_external_needs but the link in the first repo1 document isderived by link has to be also generated could you help me with how to achieve this.

**Comment by danwos** [comment]
After the build of Project B, you can use its need.json to regenerate Project A and build the HTML output. This means project A has to be built twice.

Full workflow:
1. Build ``needs.json`` of project A (`make needs`).
2. Configure project B to use external needs and to build `needs.json` (needs_build_json)
3. Build project B (`make html`). Project B contains all links.
4. Configure project A to use external needs.
5. Build project A (`make html`). Project A contains all links.

**Comment by vinodreddy-g** [comment]
While searching for some solutions , i came across this .
https://github.com/sphinx-doc/sphinx/issues/12055

Is there also a plan to create like an inventory builder.

**Reply by danwos** [reply]
I think the `needs.json` file is our inventory.
And a `make needs` creates just this file, without any calculation for data and files of HTML or PDF outputs. So a `make needs` runs much faster than a `make html`.

**Comment by David-Le-Nir** [comment]
my 2 cents:

The solution we advise to our users is to create a third document in which they reference the 2 other documents and then create 2 uni-directional matrix

We additional dynamic function, for example, you can even make some data aggregation to display more useful data in your matrix

We do not create bidirectional matrixes as it introduce a kind of infinite loop:

1. publish docA_v1
2. then refer it from docB_v1 and publish it
3. then build a docA_v1.1 referring to docB_v1
4. but now docB is not aligned on docA ...
